lib/logstash/filters/csv.rb

```diff
@@ -120,7 +124,13 @@ def filter(event)
 
     if (source = event.get(@source))
       begin
-        values = CSV.parse_line(source, :col_sep => @separator, :quote_char => @quote_char)
+
+        values = CSV.parse_line(source, :col_sep => @separator, :quote_char => @quote_char)        
+
+        if(@skip_empty_rows && values.nil?)
+          event.cancel
```

Comment:
You should not cancel the event in this case, it will prevent the event from being handled by subsequent filters and outputs.

We need to be mindful of whether the event has other non-empty fields or not. If the user is applying the csv filter to just one field of many then they probably want to process the event some more, but if they only have one field with csv data and that field is empty then they will want to `drop` the event in the next filter. We are not able to guess the users wishes here.

My advice:
- add a tag `_csvskippedemptyfield` which the user can use to construct a if conditional drop section around.
- add some more comments here and in the option declaration (line 50) to explain and advise of the `if drop`.
- change the test to verify that the tag is in place and that the event is not cancelled using `event.cancelled?`